src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -23,34 +23,45 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comments = requestToString(request);
```

Comment:
Instead of passing null through the whole servlet, having all methods need to handle null, and eventually returning an empty JSON object (potentially having the client confused why it returned an empty object), you could catch the syntax error, and return an HTTP error explaining to the client that something went wrong (this case JSON syntax error).

This way the client knows what went wrong and you don't have to handle null values everywhere.